### DISCUSSION TITLE
How to convert to triangle

### DISCUSSION THREAD

**Discussion by Dayananda02** [casact/chainladder-python#229] [Q&A]
Please let us know how to convert dataframe to triangle to make use of methods of triangle.

**Comment by jbogaardt** [comment]
This is covered in great depth in the docs:
https://chainladder-python.readthedocs.io/en/latest/tutorials/triangle-tutorial.html

Happy to answer a more specific question if anything is unclear.

**Reply by ninahjelm** [reply]
Hi, I have read this documentation, but not sure it addresses this - is there a way to use triangles already formatted in excel/csv as a triangle (not flat database) in CL python package? I can of course import them as data frames, but I want them to be triangle objects to use the triangle methods

**Reply by jbogaardt** [reply]
The package currently requires data to be tidy/long format when making a `Triangle`. If your data is already in the form of a triangle shape, you can refer to #218 on an example on how to adapt it.
